mne/viz/_figure.py

```diff
@@ -0,0 +1,1854 @@
+# -*- coding: utf-8 -*-
+"""Figure classes for MNE-Python's 2D plots."""
+
+# Authors: Daniel McCloy <dan@mccloy.info>
+#
+# License: Simplified BSD
+
+import platform
+from copy import deepcopy
+from functools import partial
+from collections import OrderedDict
+import numpy as np
+from matplotlib.figure import Figure
+from .utils import (plt_show, plot_sensors, _setup_plot_projector, _events_off,
+                    _set_window_title, _merge_annotations, DraggableLine,
+                    _get_color_list)
+from ..utils import set_config
+from ..annotations import _sync_onset
+from ..io.pick import (channel_indices_by_type, _DATA_CH_TYPES_SPLIT,
+                       _DATA_CH_TYPES_ORDER_DEFAULT)
+
+
+class MNEFigParams:
+    """Container object for MNE figure parameters."""
+
+    def __init__(self, **kwargs):
+        # default key to close window
+        self.close_key = 'escape'
+        vars(self).update(**kwargs)
+
+
+class MNEFigure(Figure):
+    """Base class for 2D figures & dialogs; wraps matplotlib.figure.Figure."""
+
+    def __init__(self, **kwargs):
+        # figsize is the only kwarg we pass to matplotlib Figure()
+        figsize = kwargs.pop('figsize', None)
+        super().__init__(figsize=figsize)
+        # add our param object
+        self.mne = MNEFigParams(**kwargs)
+
+    def _close(self, event):
+        """Handle close events."""
+        # remove references from parent fig to child fig
+        is_child = getattr(self.mne, 'parent_fig', None) is not None
+        is_named = getattr(self.mne, 'fig_name', None) is not None
+        if is_child:
+            self.mne.parent_fig.mne.child_figs.remove(self)
+            if is_named:
+                setattr(self.mne.parent_fig.mne, self.mne.fig_name, None)
+
+    def _keypress(self, event):
+        """Handle keypress events."""
+        if event.key == self.mne.close_key:
+            from matplotlib.pyplot import close
+            close(self)
+        elif event.key == 'f11':  # full screen
+            self.canvas.manager.full_screen_toggle()
```

Comment:
Why not test this one? Does it throw an error on `Agg`? If not but it's a no-op at the mpl end, it's probably still worth "testing" it to hit the branch in our code.

Reply:
It is tested in `test_plot_raw_keypresses`:

https://github.com/drammock/mne-python/blob/18de83aa6b291cf97db26572506a80078c7ebb0e/mne/viz/tests/test_raw.py#L365-L366

I don't know why Codecov thinks it's not tested.